assignment_handlers_test.go

```
@@ -0,0 +1,592 @@
+
+func (m *mockAssignmentListerByVehicle) ListAssignmentsByVehicle(ctx context.Context, vehicleID string) ([]AssignmentResponse, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	return m.assignments, nil
+}
+
+// --- helpers ---
+
+func postAssignment(handler http.HandlerFunc, body []byte, contentType string) *httptest.ResponseRecorder {
+	req := httptest.NewRequest("POST", "/api/v1/admin/assignments", bytes.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)